tests/test_run_ref.py

```diff
@@ -0,0 +1,295 @@
+            with open(fname, "wb") as f:
+                f.write(mwa_beam_response.content)
+
+
+def compare_uvh5(uv_ref, uv_new):
+    # takes as input two UVData objects, and computes relevant quantities for determining how
+    # similar the data are. Prints the histories before setting them equal. Currently only runs
+    # an equality check but should make a much more exhaustive check OR just turn
+    # back on the exact check and update the sim output when it differs
+    import numpy as np
+
+    # print histories
+    print("History 1:")
+    print(uv_ref.history)
+    print("")
+
+    print("History 2:")
+    print(uv_new.history)
+    print("")
+
+    # set history to match so equality check doesn't fail
+    uv_new.history = uv_ref.history
+
+    ref_arr, new_arr = uv_ref.data_array, uv_new.data_array
+
+    # mean diff
+    mean_diff_of_vis = np.mean(np.abs(new_arr - ref_arr))
+    mean_diff_of_abs = np.mean(np.abs(ref_arr) - np.abs(new_arr))
+
+    # compute max absolute difference and max relative difference
+    # should be identical to the allclose output for maximum absolute
+    # and relative difference
+    max_absolute_diff = np.amax(np.abs(new_arr - ref_arr))
+    frac_diff = frac_diff = np.abs(new_arr - ref_arr) / np.abs(ref_arr)
+    frac_diff[np.isnan(frac_diff)] = 0
+    max_relative_diff = np.amax(frac_diff)
+
+    # generate a true/false ndarray for passing and failing cases
+    # should match output of np.testing.assert_allclose
+    cases = np.abs(new_arr - ref_arr) <= (1e-8 + 1e-5 * np.abs(ref_arr))
+    outcome = cases.all()
+
+    # get unique outcomes (true / false) and corresponding counts
+    # then convert to dict and get result
+    unique, counts = np.unique(cases, return_counts=True)
+    outcome_dict = dict(zip(unique, counts, strict=False))
+
+    # need to check that key exists
+    if False in outcome_dict:
+        num_mismatched = str(outcome_dict[False]) + "/" + str(cases.size)
+    else:
+        num_mismatched = "0" + "/" + str(cases.size)
+
+    # print some things for reference
+    print(
+        f"mean of abs of diff of visibilities "
+        f"'mean(abs(old_data_arr - new_data_arr))': {mean_diff_of_vis}"
+    )
+    print(
+        f"mean of diff of abs of visibilities "
+        f"'mean(abs(old_data_arr) - abs(new_data_arr))': {mean_diff_of_abs}"
+    )
+    print(f"max_absolute_diff: {max_absolute_diff}")
+    print(f"max_relative_diff: {max_relative_diff}")
+    print(f"outcome: {outcome}")
+    print(f"num_mismatched: {num_mismatched}")
+
+    # perform equality check between historical and current reference
+    # simulation output
+    assert uv_new == uv_ref
+
+    # fail if any element of data_array is different between new and ref
+    # currently commented out but might be worth implementing and updating ref_sim file each time
+    # the assertion fails for the relevant files
+    # assert (ref_arr == new_arr).all()
```

Comment:
I don't think we want an exact equality check for floating point numbers, but I'd be comfortable with pretty tight constraints. The defaults in `np.testing.assert_allclose` might be a good choice, but you might also have a good feel for what that should be.

Reply:
I have updated the code to use the defaults listed in `np.testing.assert_allclose`: atol=0, rtol=1e-7. I have also removed the commented out exact equality check, and removed the discussion about it at the top of the method.